### PR TITLE
Items sf

### PR DESCRIPTION
**Pull request by StephanieFriend**
- Created methods increment and decrement in the cart controller
- Created logic in the show page for the increment and decrement in the cart
- Adjusted item display page to not show disabled items unless a merchant
- Created logic for the visitor checkout
- Created and altered tests for all logic
- All tests are passing

### CODE REVIEW

**Comment by StephanieFriend** [comment]
Pulled down Daniel's branch. No merge conflicts

**Comment by StephanieFriend** [comment]
Pulled and pushed Margo's PR

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -22,6 +22,34 @@ def remove_item
...
+          cart.contents[item_id] += 1
+        else
+          flash[:inventory_notice] = "Do not have enough of this item in stock, please choose another."
+        end
+      end
+    end
+    redirect_to '/cart'
+  end
+
+  def decrement
```
Comment (margoflewelling) [inline app/controllers/cart_controller.rb:39]: Nice logic. Like dan said maybe we need to look at being more restful at some point.

File: app/views/cart/show.html.erb [hunk 2]
```diff
@@ -30,7 +30,10 @@
       </td>
       <td>
         <center>
-          <p><%=quantity%></p>
+          <p><%=quantity%>
+            <%= button_to "+", "/cart/#{item.id}/increment", form: {style: 'height: 30px; width: 0px; padding: 0; margin: 0;'}, method: :patch %>
```
Comment (margoflewelling) [inline app/views/cart/show.html.erb:34]: ooh fancy styling

File: spec/features/items/statistic_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      click_button "+"
+      click_button "+"
+      expect(page).to have_content(7)
+    end
+
+    within "#cart-item-#{@pens.id}" do
+      expect(page).to have_content(1)
+    end
+
+    within "#cart-item-#{@gloves.id}" do
```
Comment (margoflewelling) [inline spec/features/items/statistic_index_spec.rb:87]: So much adding!

**Pull request merged**